port/common/omrstr.c

```diff
@@ -1417,7 +1417,9 @@ populateWithDefaultTokens(struct OMRPortLibrary *portLibrary, struct J9StringTok
 
 	/* Let's add the username, note that this is done seperately since it
 	 * may fail on some platforms, but in that event, we just don't add it */
```

Comment:
We should also update the comment with the details of the new approach.

Reply:
> The commit message does not match the PR's description. Should they match?

Yeah, updated.

> Is USER env-var available on all platforms?
Will USER always contain a valid value?
Can USER be greater than 128 bytes? USERNAME_BUF_LEN is set to 128. If a larger buffer is required, sysinfo_get_env will return the actual number of bytes required. Do we want to resize the buffer and store the larger value for USER?

From my experiments, it is available at `AIX`, `Linux`, `macOS`, `Windows`, and `zOS` when login via a console. The expectation is that it contains a valid value.
If the `USER` is longer than `USERNAME_BUF_LEN`, the existing behaviour is to skip it, we probably don't want an arbitrarily long string added to the token.

Updated the comment accordingly.

@babsingh this is ready for another look.

Reply:
can we just remove the call to `sysinfo_get_username`

Reply:
UID is not essential and it is allowed to fail

Reply:
we have no control over the criu app environment

Reply:
is there a way to add a dynamic check for CRIU? perhaps with portGlobals?

Reply:
> is there a way to add a dynamic check for CRIU? perhaps with portGlobals?

See `PPG_sysinfoControlFlags`. We can rename it to be used from `omrstr`: `PPG_globalControlFlags`. The new flag can be set in OpenJ9 explicitly for CRIU: `j9port.c::j9port_startup_library`. By default, the flag will be disabled in OMR.

Reply:
> is there a way to add a dynamic check for CRIU? perhaps with portGlobals?

How about move https://github.com/eclipse-openj9/openj9/blob/33900fb71ea13154b5a0834ff7c46a05b783d130/runtime/oti/j9port_generated.h#L391-L400
```
	/* The delta between Checkpoint and Restore of j9time_current_time_nanos() return values.
	 * It is initialized to 0 before Checkpoint, and set after restore.
	 * Only supports one Checkpoint, could be restored multiple times.
	 */
	int64_t nanoTimeMonotonicClockDelta;
	/* Invoking j9sysinfo_get_username()/getpwuid() with SSSD enabled can cause checkpoint failure.
	 * It is safe to call those methods if checkpoint is disallowed after a final restore.
	 * https://github.com/eclipse-openj9/openj9/issues/15800
	 */
	BOOLEAN finalRestore;
```
 into https://github.com/eclipse/omr/blob/fc60df565a221c14ba04f69dc9dcf17c828d3a5b/include_core/omrport.h#L1808

Reply:
the above comment is incomplete. yes, `finalRestore` can be moved and converted into a bit-set.

Reply:
Introduced `PPG_instantOnFlags`(``portLibrary->portGlobals->platformGlobals.instantOnFlags`) to store `OMRPORT_INSTANTON_ENABLED` & `OMRPORT_INSTANTON_FINAL_RESTORE`.
When adding `omrportpriv.h`, discovered a few method signature mismatching within `omrstr.c`, and updated the method implementation w/o changing `OMRPortLibrary` defines.

@babsingh this is ready for another round.

Reply:
> D:\a\1\s\port\common\omrstr.c(1433,6): error C2065: 'PPG_instantOnFlags': undeclared identifier [D:\a\1\s\build\port\omrport_obj.vcxproj]
D:\a\1\s\port\common\omrstr.c(1434,6): error C2065: 'PPG_instantOnFlags': undeclared identifier [D:\a\1\s\build\port\omrport_obj.vcxproj]
 
Looking at this windows failure.